app/domain/search/result.rb

```diff
@@ -14,5 +15,35 @@ def content_items
     def options_for(identifier)
       filter_options.fetch(identifier)
     end
+
+    def next_content_item(content_item)
+      next_scope =
+        sort.where_only_after(
+          content_items.unscope(:limit, :offset),
+          content_item
+        )
+
+      limit = 10
+
+      loop do
+        ids = next_scope.limit(limit).pluck(:id)
+        current_item_index = ids.index(content_item.id)
+
+        if current_item_index.nil?
+          limit *= 2
+          # Give up if the limit is greater than 500, as this process
+          # is probably taking too long, and might not even finish if
+          # somehow the content item provided has fallen out of the
+          # filter criteria
+          return false if limit > 500
+          next
+        elsif limit == (current_item_index + 1)
+          limit += 1
+          next
+        else
+          return next_scope.find_by(id: ids[current_item_index + 1])
+        end
+      end
+    end
```

Comment:
I'm a little concerned this approach is quite complicated. It might be difficult for someone new to the code to understand how this works. I wonder if we could extract this behaviour into something that's unit tested? Or maybe there's a simpler way to do this?